classes/Absences/class.xaliAbsenceStatementFormGUI.php

```diff
@@ -1,150 +1,175 @@
-		$lng = $DIC['lng'];
-		$ilCtrl = $DIC['ilCtrl'];
-		$this->absence_statement = $xaliAbsenceStatement;
-		$this->parent_gui = $parent_gui;
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\FileUpload\FileUpload;
+use ILIAS\HTTP\Wrapper\WrapperFactory;
+
+class xaliAbsenceStatementFormGUI extends ilPropertyFormGUI
+{
+    protected ilAttendanceListPlugin|ilPlugin $pl;
+    protected ilLanguage $lng;
+    protected ilCtrl $ctrl;
+    protected FileUpload $upload;
+    protected ?xaliAbsenceStatement $absence_statement;
+    private xaliAbsenceStatementGUI $parent_gui;
+    private WrapperFactory $httpWrapper;
+
+    public function __construct(xaliAbsenceStatementGUI $parent_gui, xaliAbsenceStatement $xaliAbsenceStatement)
+    {
+        global $DIC;
+        $lng = $DIC->language();
+        $ilCtrl = $DIC->ctrl();
+        $this->absence_statement = $xaliAbsenceStatement;
+        $this->parent_gui = $parent_gui;
         /** @var $component_factory ilComponentFactory */
         $component_factory = $DIC['component.factory'];
         /** @var $plugin ilAttendanceListPlugin */
-        $this->pl  = $component_factory->getPlugin(ilAttendanceListPlugin::PLUGIN_ID);
-		$this->lng = $lng;
-		$this->ctrl = $ilCtrl;
-		$this->upload = $DIC->upload();
-
-		parent::__construct();
-
-		if ($file_id = $this->absence_statement->getFileId()) {
-			$this->ctrl->setParameter($this->parent_gui, 'file_id', $file_id);
-		}
-		$this->ctrl->setParameter($this->parent_gui, 'back_cmd', $_GET['back_cmd']);
-		$this->ctrl->setParameter($this->parent_gui, 'entry_id', $xaliAbsenceStatement->getEntryId());
-		$this->setFormAction($this->ctrl->getFormAction($this->parent_gui));
-		$this->initForm();
-		$date = xaliChecklist::find(xaliChecklistEntry::find($xaliAbsenceStatement->getEntryId())->getChecklistId())->getChecklistDate();
-		$this->setTitle(sprintf($this->pl->txt('form_title_absence'), $date));
-	}
-
-
-	protected function initForm(): void
+        $this->pl = $component_factory->getPlugin(ilAttendanceListPlugin::PLUGIN_ID);
+        $this->lng = $lng;
+        $this->ctrl = $ilCtrl;
+        $this->upload = $DIC->upload();
+        $this->httpWrapper = $DIC->http()->wrapper();
+
+        parent::__construct();
+
+        if ($file_id = $this->absence_statement->getFileId()) {
+            $this->ctrl->setParameter($this->parent_gui, 'file_id', $file_id);
+        }
+
+        $backCmd = $this->httpWrapper->query()->retrieve(
+            "back_cmd",
```

Comment:
Pls unify the quotation (here and in all other occurences).

If single or double quote i leave to your preference but it should be a unified quotation approach.